homeassistant/components/srp_energy/sensor.py

```diff
@@ -0,0 +1,148 @@
+"""Support for SRP Energy Sensor."""
+from datetime import datetime, timedelta
+import logging
+
+import async_timeout
+from requests.exceptions import ConnectionError as ConnectError, HTTPError, Timeout
+
+from homeassistant.const import ATTR_ATTRIBUTION
+from homeassistant.helpers import entity
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    ATTRIBUTION,
+    DEFAULT_NAME,
+    DOMAIN,
+    ENERGY_KWH,
+    ICON,
+    MIN_TIME_BETWEEN_UPDATES,
+    SENSOR_NAME,
+    SENSOR_TYPE,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the SRP Energy Usage sensor."""
+    # API object stored here by __init__.py
+    api = hass.data[DOMAIN]
+
+    async def async_update_data():
+        """Fetch data from API endpoint.
+
+        This is the place to pre-process the data to lookup tables
```

Comment:
can remove this now (repeat from scaffold comments)

Reply:
@dgomes I've removed what you listed and made the changes you asked for. Anything else you see?